src/Field.php

```diff
@@ -497,6 +507,42 @@ public function setNull(): self
         return $this;
     }
 
+    private function getDatabasePlatform(): Platforms\AbstractPlatform
+    {
+        return $this->getOwner()->persistence !== null
```

Comment:
Should we should typecast with one generic platform - let say Sqlite?

All results will be then same and we can implemented 2 static methods for typecasting back and from like `Field::convertToUnmanagedValue($value)` and `Field::convertToPhpValue($value)`. Which can be easily used also for defining Scope values etc.

Or take care of this inside dsql Expression?

Reply:
We have to keep the persistence agnostic nature of ATK.
My idea is that ATK specific types are introduced and when typecasting is done field specific or persistence specific then persistence maps this type to its own storage mechanism.

For instance we have ATK `Type\Array`. For Array persistence there will be no typecasting but for Sql for instance it will be typecasted to `Type\Text`.
So for this field the Sql persistence knows it must store a Type\Text

Still rough - working on the concept

Reply:
`Type\Array` should typecast to string (json, serialize, ...) and it can work everywhere. I think, it should always typecast to unmanaged type, ie. objects should be newly recreated.

Reply:
I dropped these new methods completely, just use persistence with the original names.